src/meta_ops.rs

```diff
@@ -735,7 +736,113 @@ pub fn less_equal<'gc>(
     lhs: Value<'gc>,
     rhs: Value<'gc>,
 ) -> Result<MetaResult<'gc, 2>, MetaOperatorError> {
-    meta_metaop(ctx, lhs, rhs, MetaMethod::Le, |a, b| {
+    meta_metaop(ctx, lhs, rhs, MetaMethod::Le, |_, a, b| {
         Some(a.to_constant()?.less_equal(&b.to_constant()?)?.into())
     })
 }
+
+#[derive(Debug, Clone, Collect)]
+#[collect(no_drop)]
+pub struct ConcatMetaCall<'gc> {
+    pub function: Function<'gc>,
+    // Needs to be owned to work around borrowing issues in
+    // LuaFrame::call_meta_function
+    pub args: Vec<Value<'gc>>,
+}
+
+#[derive(Debug, Clone, Collect)]
+#[collect(no_drop)]
+pub enum ConcatMetaResult<'gc> {
+    Value(Value<'gc>),
+    Call(ConcatMetaCall<'gc>),
+}
+
+pub fn concat<'gc>(
+    ctx: Context<'gc>,
+    values: &[Value<'gc>],
+) -> Result<ConcatMetaResult<'gc>, MetaOperatorError> {
```

Comment:
This is a total bikeshed but personally I prefer `concat_single` -> `concat` and `concat` -> `concat_many`, making the two argument concat the "fundamental" one.

Reply:
Fixed in the latest version.